pygmt/datasets/samples.py

```diff
@@ -290,3 +217,140 @@ def _load_earth_relief_holes():
+    description : str
+        The description of the sample dataset.
+    """
+
+    func: Callable
+    description: str
+
+
+datasets = {
+    "bathymetry": GMTSampleData(
+        func=_load_baja_california_bathymetry,
+        description="Table of ship bathymetric observations off Baja California",
+    ),
+    "earth_relief_holes": GMTSampleData(
+        func=_load_earth_relief_holes,
+        description="Regional 20 arc-minutes Earth relief grid with holes",
+    ),
+    "fractures": GMTSampleData(
+        func=_load_fractures_compilation,
+        description="Table of hypothetical fracture lengths and azimuths",
+    ),
+    "hotspots": GMTSampleData(
+        func=_load_hotspots,
+        description="Table of locations, names, and symbol sizes of hotpots from "
+        "Mueller et al., 1993",
+    ),
+    "japan_quakes": GMTSampleData(
+        func=_load_japan_quakes,
+        description="Table of earthquakes around Japan from NOAA NGDC database",
+    ),
+    "mars_shape": GMTSampleData(
+        func=_load_mars_shape,
+        description="Table of topographic signature of the hemispheric dichotomy of "
+        "Mars from Smith and Zuber (1996)",
+    ),
+    "maunaloa_co2": GMTSampleData(
+        func=_load_maunaloa_co2,
+        description="Table of CO2 readings from Mauna Loa",
+    ),
+    "notre_dame_topography": GMTSampleData(
+        func=_load_notre_dame_topography,
+        description="Table 5.11 in Davis: Statistics and Data Analysis in Geology",
+    ),
+    "ocean_ridge_points": GMTSampleData(
+        func=_load_ocean_ridge_points,
+        description="Table of ocean ridge points for the entire world",
+    ),
+    "rock_compositions": GMTSampleData(
+        func=_load_rock_sample_compositions,
+        description="Table of rock sample compositions",
+    ),
+    "usgs_quakes": GMTSampleData(
+        func=_load_usgs_quakes,
+        description="Table of earthquakes from the USGS",
+    ),
+}
+
+
+def fmt_dataset_list(module_func):
+    """
+    Decorator to replace the ``{dataset_list}`` placeholder in the docstring.
+
+    Parameters
+    ----------
+    module_func : function
+        The function to decorate.
+
+    Returns
+    -------
+    module_func
+        The same *module_func* but with the docstring formatted.
+    """
+    texts = "\n        ".join(
+        f'- ``"{name}"``: {dataset.description}.' for name, dataset in datasets.items()
+    )
+
+    docstrings = module_func.__doc__.format(dataset_list=texts)
+    module_func.__doc__ = textwrap.dedent(docstrings)
+    return module_func
+
+
+def list_sample_data():
+    """
+    Report datasets available for tests and documentation examples.
+
+    Returns
+    -------
+    dict
+        Names and short descriptions of available sample datasets.
+
+    See Also
+    --------
+    load_sample_data : Load an example dataset from the GMT server.
+    """
+    return {name: dataset.description for name, dataset in datasets.items()}
```

Comment:
Should there be a `test_list_sample_data` that has a dictionary of all of the functions/descriptions to test line 324?

Reply:
Are you suggesting a test to check if the returned value is equal to:
```
{'bathymetry': 'Table of ship bathymetric observations off Baja California',
 'earth_relief_holes': 'Regional 20 arc-minutes Earth relief grid with holes',
 'fractures': 'Table of hypothetical fracture lengths and azimuths',
 'hotspots': 'Table of locations, names, and symbol sizes of hotpots from  Mueller et al., 1993',
 'japan_quakes': 'Table of earthquakes around Japan from NOAA NGDC database',
 'mars_shape': 'Table of topographic signature of the hemispheric dichotomy of  Mars from Smith and Zuber (1996)',
 'maunaloa_co2': 'Table of CO2 readings from Mauna Loa',
 'notre_dame_topography': 'Table 5.11 in Davis: Statistics and Data Analysis in Geology',
 'ocean_ridge_points': 'Table of ocean ridge points for the entire world',
 'rock_compositions': 'Table of rock sample compositions',
 'usgs_quakes': 'Table of global earthquakes from the USGS'}
```

Reply:
In 518900d6d9de9034345f6da744cd45d066e00493, I added a test to check if the returned value is in `dict` type.

Reply:
I think that's fine; I was just going off of the codecov alert saying that like 324 was untested. My thought had been to use something like a list of the dataset names being equal to `dict.keys()` to avoid a very lengthy comparison, but I think your way works better.